Created services files

pbbllo opened the pull request.

spicollys committed:
Merge pull request #2 from pbbllo/views

Presentation Screen

Commit:
feat(models, services) Created services files

List of Changes:
- Added requeriments packages in pubspec.yaml
- Created user model and service files
- Created Firebase service files
- Created Error and Log controllers

spicollys commented on lib/controllers/error_controller.dart:
```diff
@@ -0,0 +1,86 @@
+    late String errorLog;
+
+    switch (error.code) {
+      case "email-already-in-use":
```
Please add all application strings to utils.constants.AppStrings, this turns way easier the string reutilization and refactoring. In this special case, an Enum could be used as well.

pbbllo committed:
feat(models, services) Created services files

List of Changes:
- Added requeriments packages in pubspec.yaml
- Created user model and service files
- Created Firebase service files
- Created Error and Log controllers
- Created error message constant in utils folder
  - enum for SignOut errors
  - class for different types of erros
